signac/synced_collections/validators.py

```diff
@@ -112,10 +107,8 @@ def json_format_validator(data):
         for value in data:
             json_format_validator(value)
     elif switch_type == "NUMPY":
-        if numpy.iscomplex(data).any():
-            raise TypeError(
-                "NumPy object with complex value(s) is not JSON serializable"
-            )
+        if _is_numpy_scalar(data.item()):
```

Comment:
I see that the "complex" check was removed here, but I don't see a corresponding place where that was copied elsewhere. Complex values _do_ have corresponding Python types but are not JSON-encodable.

Reply:
Yes, and for that reason this validator should not throw a numpy-related error. Complex types will instead fall into the generic `else` clause and print `TypeError: Object of type complex is not JSON serializable`. Do you think that's insufficient or confusing? Does knowing that change what you think the error in your previous comment should say?

Reply:
I'm happy as long as some error is raised, regardless of the source. This can be resolved, it seems.